src/internal/kopia/cleanup_backups.go

```diff
@@ -188,31 +194,57 @@ func cleanupOrphanedData(
 			delete(toDelete, manifest.ID(ssid))
 
+			// This is a little messy to have, but can simplify the logic below.
+			// The state of tagging in corso isn't all that great right now and we'd
+			// really like to consolidate tags and clean them up. For now, we're
+			// going to copy tags that are related to Reasons for a backup from the
+			// item data snapshot to the backup model. This makes the function
+			// checking if assist backups should be garbage collected a bit easier
+			// because now they only have to source data from backup models.
+			if err := transferTags(d, &bm); err != nil {
+				logger.CtxErr(ctx, err).Errorw(
+					"transferring legacy tags to backup model",
+					"snapshot_id", d.ID,
+					"backup_id", bup.ID)
+
+				// Continuing here means the base won't be eligible for old assist
+				// base garbage collection or as a newer merge base timestamp.
+				//
+				// We could add more logic to eventually delete the base if it's an
+				// assist base. If it's a merge base then it should be mostly harmless
+				// as a newer merge base should cause older assist bases to be garbage
+				// collected.
+				//
+				// Either way, I don't really expect to see failures when transferring
+				// tags so not worth adding extra code for unless we see it become a
+				// problem.
+				continue
+			}
+
 			// Add to the assist backup set so that we can attempt to garbage collect
 			// older assist backups below.
 			if bup.Tags[model.BackupTypeTag] == model.AssistBackup {
-				// This is a little messy to have, but can simplify the logic below.
-				// The state of tagging in corso isn't all that great right now and we'd
-				// really like to consolidate tags and clean them up. For now, we're
-				// going to copy tags that are related to Reasons for a backup from the
-				// item data snapshot to the backup model. This makes the function
-				// checking if assist backups should be garbage collected a bit easier
-				// because now they only have to source data from backup models.
-				if err := transferTags(d, &bm); err != nil {
-					logger.Ctx(ctx).Debugw(
-						"transferring legacy tags to backup model",
-						"err", err,
-						"snapshot_id", d.ID,
-						"backup_id", bup.ID)
-
-					// Continuing here means the base won't be eligible for old assist
-					// base garbage collection. We could add more logic to eventually
-					// delete the base in question but I don't really expect to see
-					// failures when transferring tags.
+				assistBackups = append(assistBackups, &bm)
+				continue
+			}
+
+			// If it's a merge base track the time it was created so we can check
+			// later if we should remove all assist bases or not.
+			tags, err := keysForBackup(&bm)
+			if err != nil {
+				logger.CtxErr(ctx, err).Error(
+					"getting Reason keys for merge base. May keep an additional assist base")
+			}
+
+			for _, tag := range tags {
+				t, ok := mostRecentMergeBase[tag]
+				if ok && t.After(bm.CreationTime) {
```

Comment:
technically you don't need the OK check, right?  Zero valued time is the epoch.

Reply:
yea, that should work